From 22a3c0b278b10547e7dd3b20da71653a37f4059d Mon Sep 17 00:00:00 2001
From: Palmer Dabbelt <palmer@dabbelt.com>
Date: Fri, 29 Dec 2017 13:30:53 -0800
Subject: [PATCH 1/2] Add RISC-V support to crossdev

I've been working on a RISC-V Gentoo port, which lives in an overlay at
the moment.  Without this patch, when I "crossdev -t
riscv64-unknown-linux-gnu" I see

     * Host Portage ARCH:     amd64
     * Target Portage ARCH:   *

which means I don't get quite the right set of packages.

This patch adds the RISC-V tuples to crossdev's architecture mapping
list, which sets the correct target portage arch when building a RISC-V
cross environment.  With this patch, and an overlay that defines various
RISC-V ports, I can build a RISC-V cross environment on my amd64 host.
Note that I'm using crossdev-20151026 with this patch applied on my
machine.

Signed-off-by: Palmer Dabbelt <palmer@dabbelt.com>
---
 crossdev | 1 +
 1 file changed, 1 insertion(+)

diff --git a/crossdev b/crossdev
index d661c86..52dda2e 100755
--- a/crossdev
+++ b/crossdev
@@ -210,6 +210,7 @@ parse_target() {
 		mips*)      TARCH=mips;;
 		powerpc64*) TARCH=ppc64;;
 		powerpc*)   TARCH=ppc;;
+		riscv*)     TARCH=riscv;;
 		sparc*)     TARCH=sparc;;
 		s390*)      TARCH=s390;;
 		sh*)        TARCH=sh;;
-- 
2.11.0

